bin/server.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2020 Dmitry Ivanko. All Rights Reserved.
+# Copyright (C) 2020 Vladislav Yarovoy. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Server-side of Helpik. """
+
+import logging
+
+import aiohttp
+import bs4
+from aiohttp import web
+from aiohttp.web_request import Request
+
+from bin import config
+
+logging.basicConfig(filename=config.LOG_PATH, level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+
+class Handler(web.View):
+    """Class-based handler of the requests for wiki synopses. """
+
+    def __init__(self, request: Request):
+        super().__init__(request)
+
+        self._params = {
+            'action': 'parse',
+            'prop': 'text',
+            'page': '',
+            'origin': '*',
+            'format': 'json',
+            'formatversion': 2
+        }
+        self._data = ''
+        self._client_language = config.DEFAULT_LANGUAGE
+
+    async def get(self):
+        """Returns content of a page by a page name. """
+
+        synopsis = {}
+
+        if 'pageName' in self.request.query:
+            self._params['page'] = self.request.query['pageName']
+
+        if 'language' in self.request.query:
+            self._client_language = self.request.query['language']
+
+        await self.get_synopsis()
+
+        synopsis['text'] = self._content_parser()
+        synopsis['url'] = config.MEDIAWIKI_API_URL + self._params['page']
+        LOGGER.info('Response to the client`s request')
+
+        if synopsis['text'] == 'None':
+            return web.StreamResponse(status=404)
+
+        return web.json_response(synopsis)
+
+    async def get_synopsis(self):
+        """Parses the content of the wiki page and fetches synopsis. """
+
+        self._params['page'] = f"{self._params['page']}/{self._client_language}"
+        content = await self._request_mediawiki_api()
+
+        if content == '':
+            LOGGER.error('MediaWiki not found')
+            self._data = ''
+            return self._data
```

Comment:
1) Create the module called `exceptions.py`.
2) Raise a custom exception named `CouldNotConnectToMediaWiki` instead of returning an empty string.